### ISSUE TITLE
Code modernization issues

### ISSUE THREAD

**Issue by wholder** [marytts/marytts#1007]
At this point, I've managed to update and rework quite a large section of the code.  There's more work to be done, but I've run into an issue that I think needs to be resolved.  Specifically, I see a lot of constructs like this in certain modules, such as this code from `PitchStatistics.java`:

```
  try {
    intercept = ler.readDouble();
  } catch (IOException e) {
    // TODO Auto-generated catch block
    e.printStackTrace();
  }
```
In many cases, there are long sequences of this code like this.  However, while the code catches the exception and prints a stack trace, it keeps running.  And, in most cases, the following code  will also print an error, and then continue on.  In my opinion, the code should be reworked to collapse these sequential cases into one then throw the first error encountered.  However, there are so many places where this code exists, that I wanted to get agreement on this before I proceed to changing it.

There are similar issues regarding how exceptions are handled (or not handled) in other code that I also think need to be addressed.  In some cases, the code is structured in a way that will fail to report certain errors or, in other cases, hides the fact that some blocks in the code are not reachable.  In most cases I've been able to restructure this code and then address the issues that emerge.  However, again, I wanted to let you guys know what I'm doing in case you have other thoughts.

Wayne

**Comment by psibre** [comment]
Yeah, that's the kind of crap that Eclipse auto-generated and various contributors couldn't be bothered to implement properly. In fact this specific anti-pattern is exactly what -- even back in 2010 -- @marc1s railed against in his attempts to rein in the rampant error mis-handling (see also https://github.com/marytts/marytts/pull/1003#issuecomment-643578687).

The gist of it is that any surviving instances of `e.printStackTrace()` as the only instruction in a `catch` block should be fixed. But it's also more likely to persist in unused code, and we could just as well consider removing it altogether. A lot of the code in `marytts-signalproc` has been unmaintained for many years, and even then, a lot of it was research code. Since we were using Subversion at the time, there was no concept of modularizing code repositories, and a lot of stuff that was hacked together for some research paper or other was just committed to trunk with no code review.

**Comment by wholder** [comment]
Thanks for the info.  I'll work on cleaning these up as I progress through the code.  BTW, since you mention that "a lot of the code in `marytts-signalproc` has been unmaintained for many years", it would help to know what other code, if any, is not on the critical path.  I'm running the various tests that I know about as I go along, but I want to avoid unintentionally breaking anything, so know what code might not be fully covered by tests would also be helpful.

Wayne
